### PR TITLE
Extract to parser helpers

### PR DESCRIPTION
**Pull request by timse**
**What kind of change does this PR introduce?**

refactoring

**Did you add tests for your changes?**

Is covered by existing tests

**If relevant, link to documentation update:**

N/A

**Summary**

While doing my refactorings to es6 i noticed that certain calls to the parser repeat themselves over several modules. 
First I tried to pull them as a whole into a helper function but then realized that for now a more atomic approach should probably be more reasonable, as it removes the sideeffects and makes it reusable throughout the code.

For me the these helper functions abstract a lot of the clutter away and make the code more readable as I can see with one glimpse what is happening.

**Does this PR introduce a breaking change?**

No

**Other information**

If this approach is "ok" I would love to continue on this bit by bit.
If the PR gets to big it will introduce too many conflicts with existing PRs I guess.
Therefore if I may continue, I would try this with a bunch of smaller PRs to keep the impact on files and therefore other PRs as small as possible.

### CODE REVIEW

File: lib/ParserHelpers.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ParserHelpers.toConstantDependency = function toConstantDependency(value) {
+	return function(expr) {
+		var dep = new ConstDependency(JSON.stringify(value), expr.range);
+		dep.loc = expr.loc;
+		this.state.current.addDependency(dep);
+		return true;
+	};
+};
+
+ParserHelpers.evaluateToString = function setTypeof(value) {
```
Comment (sokra) [inline lib/ParserHelpers.js]: wrong function name

File: lib/ParserHelpers.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+			}
+		},
+		module: parser.state.module
+	});
+	parser.state.current.addVariable(name, expression, deps);
+	return true;
+};
+
+ParserHelpers.toConstantDependency = function toConstantDependency(value) {
+	return function(expr) {
```
Comment (sokra) [inline lib/ParserHelpers.js]: could you name these anonymous function? It better for stacktraces and profiling...

**Comment by timse** [comment]
What still personally "annoys" me, is that the callbacks of the plugins are called with the parser as the context and then mutate the parsers state implicitly by adding the ConstDependency to it.
Is there any way we can pass the state as a parameter e.g. in Tapable? Or make Tapable a promise?
If i just return the e.g. ConstDependency created by the callback and assign it to the state it gets fairly verbose again.

Promises would probably make the "return true/false" impossible though?

File: lib/ParserHelpers.js (hunk 2 above)
Comment (timse) [inline lib/ParserHelpers.js]: done :)

File: lib/ParserHelpers.js (hunk 1 above)
Comment (timse) [inline lib/ParserHelpers.js]: done :)

**Comment by TheLarkInn** [comment]
Tapable could maybe return a promise in the future but it's kind of tough. 

Nice work on this though. This is significantly easier to read now.

**Pull request merged**